lib/ast-linter/rules/link-no-unknown-partials.js

```diff
@@ -0,0 +1,23 @@
+const Rule = require('./base');
+const {getNodeName} = require('../helpers');
+
+module.exports = class NoUnknownPartials extends Rule {
+    _checkForUnknownPartials(node) {
+        if (node.type === 'PartialStatement' && node.name) {
+            if (!this.isValidPartialReference(node)) {
+                this.log({
+                    message: `The partial ${getNodeName(node)} could not be found`,
+                    line: node.loc && node.loc.start.line,
+                    column: node.loc && node.loc.start.column,
+                    source: this.sourceForNode(node)
+                });
+            }
+        }
+    }
+
+    visitor() {
+        return {
+            PartialStatement: this._checkForUnknownPartials.bind(this)
```

Comment:
I would add `PartialBlockStatement` here to collect all partials.

The reason is in the Handlebars spec : a partial can be called as a block to provide a fallback in case the partial isn’t found (https://handlebarsjs.com/guide/partials.html#partial-blocks).